### PR TITLE
[go-client] integrated `logrus`, `testify` and `go mod` for pulsar-client-go

### PR DESCRIPTION
**Pull request by wolfstudy**
Fixes [issue:3417](https://github.com/apache/pulsar/issues/3417)

### Motivation

use go community provides some good libraries, help us save a lot of development work

### Modifications

- log:[logrus](https://github.com/Sirupsen/logrus)
- test util:[testify](https://github.com/stretchr/testify)
- go pkg management:[go mod](https://github.com/golang/go/wiki/Modules)

### CODE REVIEW

File: pulsar-client-go/pulsar/c_go_pulsar.h [hunk 1]
```diff
@@ -25,16 +25,6 @@
 
 // Callback proxy functions
 
-void pulsarClientLoggerProxy(pulsar_logger_level_t level, char* file, int line, char* message, void *ctx);
-
-static inline void pulsarClientLoggerConstProxy(pulsar_logger_level_t level, const char* file, int line, const char* message, void *ctx) {
-    pulsarClientLoggerProxy(level, (char*)file, line, (char*)message, ctx);
-}
-
-static inline void _pulsar_client_configuration_set_logger(pulsar_client_configuration_t *conf, void *ctx) {
```
Comment (merlimat) [inline pulsar-client-go/pulsar/c_go_pulsar.h]: This is used to route all the C++ library logs into a Go logger function. If we don't have this, all logs from C++ lib will be printed with a different format on stdout.

File: pulsar-client-go/examples/consumer-listener/consumer-listener.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ package main
 import (
 	"fmt"
 	"github.com/apache/pulsar/pulsar-client-go/pulsar"
-	"log"
+	log "github.com/pulsar/pulsar-client-go/logutil"
```
Comment (merlimat) [inline pulsar-client-go/examples/consumer-listener/consumer-listener.go]: should this be instead `"github.com/apache/pulsar/pulsar-client-go/logutil"`

Comment (wolfstudy) [reply pulsar-client-go/examples/consumer-listener/consumer-listener.go]: yes, we should use the full path

**Comment by sijie** [comment]
run cpp tests

**Comment by sijie** [comment]
run cpp tests

**Comment by wolfstudy** [comment]
run cpp tests

**Comment by wolfstudy** [comment]
run cpp tests

**Pull request merged**